Modules/_testcapi/watchers.c

```diff
@@ -605,28 +618,38 @@ allocate_too_many_func_watchers(PyObject *self, PyObject *args)
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+_testcapi.set_func_defaults_via_capi
+    func: object
+    defaults: object
+    /
+Set default values to a given function.
+[clinic start generated code]*/
+
 static PyObject *
-set_func_defaults(PyObject *self, PyObject *args)
+_testcapi_set_func_defaults_via_capi_impl(PyObject *module, PyObject *func,
+                                          PyObject *defaults)
+/*[clinic end generated code: output=caf0cb39db31ac24 input=9a7d91215cd0f05b]*/
 {
-    PyObject *func = NULL;
-    PyObject *defaults = NULL;
-    if (!PyArg_ParseTuple(args, "OO", &func, &defaults)) {
-        return NULL;
-    }
     if (PyFunction_SetDefaults(func, defaults) < 0) {
         return NULL;
     }
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+_testcapi.set_func_kwdefaults_via_capi
+    func: object
+    kwdefaults: object
+    /
```

Comment:
I'd clone this from `_testcapi.set_func_defaults_via_capi` and change `kwdefaults` to `defaults` in the body of the C impl function.